src/server.js

```diff
@@ -171,32 +172,51 @@ const logAndGetError = function(
     };
 };
 
-const respond400Error = (logging: Logger, res: $Response, error, value) => {
+const respond400BadRequest = (
+    logging: Logger,
+    res: $Response,
+    error,
+    value,
+) => {
     logging.error(error);
     return res.status(400).json({error, value});
 };
 
+const isValidAbsoluteURL = (str: string): boolean =>
+    typeof str === "string" && whatwgURL.parseURL(str) !== null;
```

Comment:
I don't think you need to use `whatwg-url` for this, I think you can just use the native implementation: https://nodejs.org/api/url.html#url_the_whatwg_url_api

It was added as a global in v10 but for backwards compat you can do: `const {URL} = require("url");` and then here do something like:

```
const isValidAbsoluteURL = (str: string): boolean => {
    try {
        new URL(str);
        return true;
    } catch(e) {
        return false;
    }
};
```

(`graphql-gateway` is already using `URL`)

Reply:
OK, seems reasonable